### PR TITLE
[FLINK-16038][jdbc] Add hitRate metric for JDBCLookupFunction

### PR DESCRIPTION
**Pull request by libenchao**
## What is the purpose of the change

Add hitRate metric for JDBCLookupFunction.

## Brief change log

Add a gauge metric named "hitRate" for JDBCLookupFunction.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCLookupFunction.java [hunk 1]
```diff
@@ -117,7 +118,11 @@ public void open(FunctionContext context) throws Exception {
 			this.cache = cacheMaxSize == -1 || cacheExpireMs == -1 ? null : CacheBuilder.newBuilder()
 					.expireAfterWrite(cacheExpireMs, TimeUnit.MILLISECONDS)
 					.maximumSize(cacheMaxSize)
+					.recordStats()
 					.build();
+			if (cache != null) {
+				context.getMetricGroup().gauge("hitRate", (Gauge<Double>) () -> cache.stats().hitRate());
```
Comment (wuchong) [inline flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCLookupFunction.java]: Maybe `cacheHitRate` is more explicit?

**Comment by libenchao** [comment]
> Looks good to me. I'm just wondering whether there is a way to verify it.

I also tried to verify it in tests, bug failed to find a good way to do this.
I can verify this in internal jobs before we merge it. I'll give you feedback later.

**Comment by libenchao** [comment]
@wuchong I verified this with my internal job.
The metric is ok, however Web UI interprets all metric as `Int`, and for `cacheHitRate`, it almost always be 0. 
IMO, Web UI should not assume all metrics are `Int`, it maybe needs to open another issue to fix that.

**Comment by XuQianJin-Stars** [comment]
+1 LGTM

**Comment by MartijnVisser** [comment]
@libenchao We've now moved the code from the JDBC connector to its own dedicated repository; please re-route this PR to https://github.com/apache/flink-connector-jdbc

**Pull request closed without merging**